### PR TITLE
Use a pool of CNAMEs to fetch IPs

### PR DESCRIPTION
**Pull request by jadeallenx**
Problem to solve: As mentioned in https://github.com/helium/miner/pull/1267 we can only have a maximum of 16 IPv4 DNS addresses per CNAME for seed nodes, but we currently have significantly more than 16 seed node IPs in service. 

Solution: Use a pool of DNS CNAMEs to fetch IPs, concatentate them all, and then add the static IPs to form the complete list of seed nodes available for service.  This PR introduces a new environment variable `seed_dns_pool_names` which configures the additional CNAME record names. This variable value is expected to be a list of strings like `["seed2", "seed3", ... ]`

### CODE REVIEW

File: src/group/libp2p_group_gossip_server.erl [hunk 1]
```diff
@@ -44,6 +44,7 @@
 -define(DNS_TIMEOUT, 2000). % millis
 -define(DNS_SLEEP, 100). % millis
 -define(DEFAULT_SEED_DNS_PORTS, [2154, 443]).
+-define(SEED_CONFIG_DNS_NAME, "_seed_config.helium.io").
```
Comment (Vagabond) [inline src/group/libp2p_group_gossip_server.erl:47]: the DNS record is configutable, can we make this configurable too?

Comment (evanmcc) [reply src/group/libp2p_group_gossip_server.erl:47]: we don't seem to use this anymore now that we pull it from the app env.  should it be the default there, or is there some other place that it's used?

**Review by evanmcc — APPROVED** [review]
code looks good, but I had one question about the define

File: src/group/libp2p_group_gossip_server.erl [hunk 2]
```diff
@@ -395,31 +395,62 @@ choose_random_element([E]) -> E;
...
+    case attempt_dns_lookup(CName, ?DNS_RETRIES) of
+        {error, _} = Error ->
+            lager:error("DNS lookup of ~p resulted in ~p after ~p retries", [CName, Error, ?DNS_RETRIES]),
+            Acc;
+        {ok, DNSRecord} ->
+            lager:debug("successful DNS lookup result: ~p for ~p", [lager:pr(DNSRecord, inet), CName]),
+            convert_dns_records(DNSRecord) ++ Acc
     end.
 
+maybe_make_seed_pool_names(Base, []) -> [Base];
```
Comment (Vagabond) [inline src/group/libp2p_group_gossip_server.erl:444]: This line should look for `undefined` no?

Comment (Vagabond) [reply src/group/libp2p_group_gossip_server.erl:444]: oh, never mind.

File: src/group/libp2p_group_gossip_server.erl [hunk 3]
```diff
@@ -395,31 +395,62 @@ choose_random_element([E]) -> E;
...
+collect_dns_records(Base) ->
+    AdditionalNames = generate_seed_pool_names(application:get_env(libp2p, seed_config_dns_name, undefined)),
+    DNSNames = maybe_make_seed_pool_names(Base, AdditionalNames),
+    lists:foldl(fun do_dns_lookups/2, [], DNSNames).
+
+generate_seed_pool_names(undefined) -> [];
+generate_seed_pool_names(ConfigDnsName) ->
+    case inet_res:lookup(ConfigDnsName, in, txt) of
+        [] -> []; %% there was an error of some kind, return empty list
+        [PoolSizeStr] ->
```
Comment (Vagabond) [inline src/group/libp2p_group_gossip_server.erl:420]:
```suggestion
        [[PoolSizeStr]] ->
```

```
(blockchain_node@127.0.0.1)1> inet_res:lookup("_seed_config.helium.io", in, txt).
[["4"]]
```

**Pull request merged**